rllib/core/learner/learner.py

```diff
@@ -568,15 +569,12 @@ def remove_module(self, module_id: ModuleID) -> None:
                 param_ref = self.get_param_ref(param)
                 if param_ref in self._params:
                     del self._params[param_ref]
-                if param_ref in self._param_to_optim:
-                    optimizer = self._param_to_optim[param_ref]
-                    if optimizer in self._optim_to_param:
-                        del self._optim_to_param[optimizer]
-                    del self._param_to_optim[param_ref]
-            optim_names = list(self._name_to_optim.keys())
-            for name in optim_names:
-                if name.startswith(module_id):
-                    del self._name_to_optim[name]
+            optim_names = self._module_to_optim_name[module_id]
+            for optim_name in optim_names:
+                optim = self._name_to_optim[optim_name]
+                del self._optim_to_param[optim]
+                del self._name_to_optim[optim_name]
```

Comment:
Nonblocking nit: Can we call this `self._optim_name_to_optim` and `self._module_id_to_optim_name`? Makes it easier to understand if we ever come back to this!

Reply:
sure

Reply:
done